java_src/foam/nanos/cron/CronScheduler.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.cron;
+
+import bsh.EvalError;
+import bsh.Interpreter;
+import foam.core.ContextAwareSupport;
+import foam.core.Detachable;
+import foam.core.FObject;
+import foam.dao.*;
+import foam.mlang.MLang;
+import foam.mlang.sink.Min;
+import foam.nanos.NanoService;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.Date;
+
+public class CronScheduler
+    extends    ContextAwareSupport
+    implements NanoService
+{
+  protected MapDAO cronDAO_;
+
+  /**
+   * Gets the minimum scheduled cron job
+   *
+   * @return Date of the minimum scheduled cron job
+   */
+  private Date getMinScheduledTime() {
+    Min min = (Min) MLang.MIN(Cron.SCHEDULED_TIME);
+    cronDAO_.select(min);
+    return ((Cron) min.getValue()).getScheduledTime();
+  }
+
+  /**
+   * Runs a Cron job script
+   *
+   * @param cron the cron job to run
+   * @return the updated cron job
+   */
+  private void runCronJob(Cron cron) {
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
The first part of this method could be added to the Script class/model as runScript() so that it could be reused.